pages/blog/geo-shapes-in-crate.html

```diff
@@ -0,0 +1,223 @@
+created: 2015-12-03
+post_name: geo-shapes-in-crate
+status: publish
+post_type: post
+tags: geo_shape, geo, gis, geojson, shape
+category: developernews
+
+In [our last blogpost][1] about Crate's spatial and geographic features we
+announced the support for the `geo_point` data type including some operations.
+Since then, we have greatly improved and added a type `geo_shape` to Crate!
+
+Starting with Crate `0.53.0` we added `geo_shape` ([documentation][4]) to
+store and work with shapes of all kinds. Together with [JOINs][8]
+working with geo-spatial data now becomes much easier. First
+of all, let us look at the new data definition:
+
+## Defining the Data
+
+To define a column of type `geo_shape`, simply add it as a type:
+
+```sql
+CREATE TABLE geo_shapes.all_shapes (
+  "geometry"       geo_shape,
+  "type"           string,
+  properties       object AS (
+    "name"         string
+  )
+);
+```
+In addition to that, it is possible to add further options to customize indexing:
+
+```sql
+CREATE TABLE geo_shapes.all_shapes (
+  "geometry"       geo_shape INDEX USING geohash WITH (PRECISION='50m', distance_error_pct=0.025),
+  "type"           string,
+  properties       object AS (
+    "name"         string
+  )
+);
+```
+
+In this statement, the `geohashing` algorithm ([wikipedia][5]) is used to create
+an index of each shape with the precision parameters controlling its accuracy.
+More on that [here][4] and [here][6]. This index will 'rasterize' all data with
+the defined `PRECISION` - hence a single point becomes a square of
+a certain size. Similarly, the percentage provided in `distance_error_pct` will
+control how precise the borders of a shape are stored.
+
+## Operations
+
+### Within
+
+`Within` is an operation that allows to find out if a point (or shape) is
+contained within another shape. Since this can be very costly we provide two
+variants of this operation: Via `match` ([documentation][9]) or by using
+`within`([documentation][10]). The former uses a - fast - approximate matching
+(on a grid) which cannot be used to join tables, whereas the latter can be used
+for JOINs but will also be more expensive since it is using vectors.
+
+### Intersects
+
+If two `geo_shape` planes share any point, this operation will return true -
+and similar to `within` there's two ways to call on this operation: `match` and
+`intersects`. As before, the `match` variant is faster and less precise.
+
+### Disjoint
+
+`disjoint` is the opposite of `intersects` - it only matches shapes that do
+not share any points.
+
+
+## Use Case: Berlin
+
+As a short demonstration, we use the Google Places API to gather a few
+locations in central Berlin. After importing and indexing everything with
+Crate, we can run queries based on GIS data!
+
+### Setup
+
+We are going to use two tables, the first of which will provide information
+on 60 different localities (gathered from [Google's Places API][2]):
+
+```sql
+CREATE TABLE geo_shapes.places (
```

Comment:
@celaus I'm OK with this, but are we missing a step that imports data? Even just a small set.